### PR TITLE
[HttpFoundation] Use Symfony exception for request unexpected values

### PR DESCRIPTION
**Pull request by VincentLanglet**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Instead of https://github.com/symfony/symfony/pull/51196
Cf https://github.com/symfony/symfony/pull/51196#issuecomment-1659838136

cc @nicolas-grekas

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/ParameterBag.php [hunk 1]
```diff
@@ -184,7 +185,7 @@ public function getEnum(string $key, string $class, \BackedEnum $default = null)
         try {
             return $class::from($value);
         } catch (\ValueError|\TypeError $e) {
-            throw new \UnexpectedValueException(sprintf('Parameter "%s" cannot be converted to enum: %s.', $key, $e->getMessage()), $e->getCode(), $e);
+            throw new UnexpectedValueException(sprintf('Parameter "%s" cannot be converted to enum: "%s".', $key, $e->getMessage()), $e->getCode(), $e);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/ParameterBag.php]:
```suggestion
            throw new UnexpectedValueException(sprintf('Parameter "%s" cannot be converted to enum: %s.', $key, $e->getMessage()), $e->getCode(), $e);
```

**Review by nicolas-grekas — APPROVED** [review]
Just one minor thing and GTM.

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/HttpFoundation/ParameterBag.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Symfony/Component/HttpFoundation/ParameterBag.php]: The ci fail without this change

Comment (VincentLanglet) [reply src/Symfony/Component/HttpFoundation/ParameterBag.php]: I reverted @derrabus @nicolas-grekas

**Comment by nicolas-grekas** [comment]
Thank you @VincentLanglet.

**Pull request merged**